### PR TITLE
Move ReadConfig to LoadConfig and validate

### PR DESCRIPTION
**Pull request by andrewemeryanz**
The config.ReadConfig method reads configuration from file and
populates passed-in config structs. To ensure that the returned
configuration is valid the method now validates the loaded
configuration before returning.

To avoid ambiguity and provide only a single mechanism to load
configuration, this method has been renamed to LoadConfig.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -40,5 +43,20 @@ func ReadConfig(cfgFile string, defaultConfig *DefaultConfig, customConfig inter
 		return err
 	}
 
-	return decoder.Decode(c)
+	err = decoder.Decode(c)
+	if err != nil {
+		return err
+	}
+
+	err = validator.Validate(defaultConfig)
```
Comment (springwiz) [inline config/config.go:51]: The change needs to be complemented with validation rule configuration on the config struct. Are the validation rules part of another PR?

Comment (andrewemeryanz) [reply config/config.go:51]: Not sure what your question is. Validation will fail if children are invalid, even if the parent isn't explicitly a `Validator`.

Comment (springwiz) [reply config/config.go:51]: Just saw that the validation rules are already annotated on the config structs i.e. validate:"nonnil"`, may be got included in another PR.

**Review by springwiz — APPROVED** [review]
LGTM

File: config/config.go (hunk 1 above)
Comment (andrewemeryanz) [reply config/config.go:51]: Yes. Not sure of the history of the validation rules, but they're untouched in this change.

**Pull request merged**